lib/engine/game/g_1848/step/buy_train.rb

```diff
@@ -37,6 +37,7 @@ def buyable_trains(entity)
             # Cannot buy 2E if one is already owned, can't by non 2e if at limit
             trains_to_buy = at_train_limit?(entity) ? [] : super
             trains_to_buy = trains_to_buy.select(&:from_depot?) unless @game.can_buy_trains
+            trains_to_buy = reject_unaffordable_depot_trains(trains_to_buy, entity) if must_buy_train?(entity)
```

Comment:
Why do you need to remove these? Won't the buy action fail if they try to buy it? In practice its good to show the train you could buy after enough loans.

Reply:
bad reading of the rules on my part. Discarded trains are removed from the game.